text/1001-edgedb-install.rst

```diff
@@ -0,0 +1,298 @@
+
+
+Overview
+========
+
+The RFC proposes a new group of ``edgedb`` CLI commands under ``edgedb server``
+prefix:
+
+* ``edgedb server install`` -- install or update a specific version of the
+  EdgeDB server on the local machine;
+
+* ``edgedb server uninstall`` -- uninstall a specific version of the
+  EdgeDB server or all versions of EdgeDB from the local machine;
+
+* ``edgedb server start`` -- starts an EdgeDB server instance;
+
+* ``edgedb server status`` -- show the status of the local EdgeDB server;
+
+* ``edgedb server stop`` -- stop the given EdgeDB server instance;
+
+* ``edgedb server upgrade`` -- upgrade the specified EdgeDB server instance
+  to the new major version.
+
+* ``edgedb server prune`` -- removes upgrade backups and other unused data.
+
+Common aspects of the implementation of the above commands is discussed
+below in `design-considerations`_.
+
+
+edgedb server install
+=====================
+
+Downloads and installs a given EdgeDB server version
+(latest stable by default).
+
+Arguments
+---------
+
+``--version=<ver>``
+  specifies the major version of the server to install.
+
+``--nightly``
+  if passed, the latest nightly build from the specified version channel
+  is installed.
+
+``--user``
+  install into the user home directory instead of system-wide, does not
+  require privelege elevation (TODO: this requires relocatable
+  builds and is TBD).
+
+``--list``
+  list EdgeDB versions available for installation
+
+``--list-installed``
+  list installed EdgeDB versions
+
+``--update``
+  if specified, ``edgedb install`` will only attempt to update the existing
+  installations.
+
+``--docker``
+  Use Docker instead of downloading and installing packages directly onto
+  the user's system.  The Docker daemon must be present and accessible by
+  the user.
+
+
+Implementation
+--------------
+
+By default, ``edgedb install`` will look for the most suitable package for
+the current platform and install it using the system's package manager
+if one is available, or download and unpack a generic package for the
+OS/arch, or use Docker if avalable (and requested).
+
+
+edgedb server uninstall
+=======================
+
+Uninstalls the specified version of EdgeDB.
+
+Synopsis
+--------
+
+``edgedb server uninstall [options]``
+
+If there are multiple versions installed, either ``--all`` or
+``--version`` or ``--unused`` is required.
+
+Options
+-------
+
+``--version=<ver>``
+  Specifies the version to uninstall.  The specified server version must
+  not be currently running.
+
+``--all``
+  Uninstalls all versions of EdgeDB.
+
+``--unused``
+  Uninstalls all versions of EdgeDB that are not used in any instance.
+
+
+edgedb server start
+===================
+
+Starts an EdgeDB server instance with the specified name.
+
+Synopsis
+--------
+
+``edgedb server start [options] [<name>]``
+
+Options
+-------
+
+``<name>``
+  The name of the EdgeDB instance.  Must be unique.  If not specified,
+  the name ``default`` is used.
```

Comment:
Can I put my instance data to a directory other than `/var/lib/edgedb/data/`? If so would I be able to use the full dir name as instance name?

Reply:
You should be able to configure the parent directory (or directories) where the commands look for instances.  An arbitrarily located data dir is out of scope.  If you need that, use `edgedb-server` directly.